repo/fsrepo/defaultds.go

```diff
@@ -39,7 +39,7 @@ func openDefaultDatastore(r *FSRepo) (repo.Datastore, error) {
 	// including "/" from datastore.Key and 2 bytes from multihash. To
 	// reach a uniform 256-way split, we need approximately 4 bytes of
 	// prefix.
-	blocksDS, err := flatfs.New(path.Join(r.path, flatfsDirectory), 4)
+	blocksDS, err := flatfs.New(path.Join(r.path, flatfsDirectory), 4, !r.config.Datastore.NoSync)
```

Comment:
may be useful to do it like this, to avoid hard to read code:

```go
syncfs := !r.config.Datastore.NoSync
blocksDS, err := flatfs.New(path.Join(r.path, flatfsDirectory), 4, syncfs)
```